library/alloc/src/vec/spec_from_iter_nested.rs

```diff
@@ -23,9 +25,27 @@ where
         let mut vector = match iterator.next() {
             None => return Vec::new(),
             Some(element) => {
-                let (lower, _) = iterator.size_hint();
-                let mut vector = Vec::with_capacity(lower.saturating_add(1));
+                // Copy RawVec::MIN_NON_ZERO_CAP
+                // do -1 because we would add 1 before `with_capacity` call.
+                let min_non_zero_cap_sub_1 = match mem::size_of::<T>() {
+                    1 => 7,
+                    b if b <= 1024 => 3,
+                    _ => 0,
+                };
```

Comment:
maybe turn this into a function so it doesn't get out of sync?

Reply:
Ok, I would just make this value public.
It is a constant.

Reply:
Done.